src/metpy/calc/boundarylayer.py

```diff
@@ -0,0 +1,305 @@
+        idxfoot=idxfoot,
+        ustar=ustar,
+    )
+
+    height = height[~np.isnan(bRi)]
+    bRi = bRi[~np.isnan(bRi)]
+
+    if any(bRi > bri_threshold):
+        iblh = np.where(bRi > bri_threshold)[0][0]
+        blh = height[iblh]
+    else:
+        blh = np.nan * units.meter
+
+    return blh
+
+
+def blh_from_parcel(
+    height,
+    potential_temperature,
+    smoothingspan: int = 5,
+    theta0=None,
+):
+    """Calculate atmospheric boundary layer height with the parcel method.
+
+    It is the height where the potential temperature profile reaches its
+    foot value.
+    See [Sei00, HL06, Col14].
+
+    Parameters
+    ----------
+    height : `pint.Quantity`
+        Altitude (metres above ground) of the points in the profile
+    potential_temperature : `pint.Quantity`
+        Potential temperature profile
+    smoothingspan : int, optional
+        The amount of smoothing (number of points in moving average)
+    theta0 : `pint.Quantity`, optional
+        Value of theta at the foot point (skip unstruted points or add extra term). If not provided, theta[0] is taken.
+
+    Returns
+    -------
+    blh : `pint.Quantity`
+        Boundary layer height estimation
+    """
+    potential_temperature = smooth(potential_temperature, smoothingspan)
+
+    if theta0 is None:
+        theta0 = potential_temperature[0]
+
+    if any(potential_temperature > theta0):
+        iblh = np.where(potential_temperature > theta0)[0][0]
+        blh = height[iblh]
+    else:
+        blh = np.nan * units.meter
+
+    return blh
+
+
+def blh_from_humidity_gradient(
+    height,
+    humidity,
+    smoothingspan: int = 5,
+    idxfoot: int = 0,
+):
+    """Calculate atmospheric boundary layer height from the relative
+    humidity gradient
+
+    It is the height where the relative humidity or specific humidity gradient reaches a minimum.
+    See [Sei00, HL06, Col14].
+
+    Parameters
+    ----------
+    height : `pint.Quantity`
+        Altitude (metres above ground) of the points in the profile
+    humidity : `pint.Quantity`
+        Humidity (relative or specific) profile
+    smoothingspan : int, optional
+        The amount of smoothing (number of points in moving average)
+    idxfoot : int, optional
+        The index of the foot point (first trusted measure), defaults to 0.
+
+    Returns
+    -------
+    blh : `pint.Quantity`
+        Boundary layer height estimation
+    """
+
+    dRHdz = mpcalc.first_derivative(smooth(humidity, smoothingspan), x=height)
+
+    dRHdz = dRHdz[idxfoot:]
+    height = height[idxfoot:]
+
+    iblh = np.argmin(dRHdz)
+
+    return height[iblh]
+
+
+def blh_from_temperature_inversion(
+    height,
+    temperature,
+    smoothingspan: int = 5,
+    idxfoot: int = 0,
+):
+    """Calculate atmospheric boundary layer height from the inversion of
+    absolute temperature gradient
+
+    It is the height where the temperature gradient (absolute or potential) changes of sign.
```

Comment:
From what I remember (my experience with this is now a bit old), this method is more suited for nocturnal stable boundary layers as it will track the end of the stable layer at the surface.

For convective boundary layer, the parcel method should be preferred to this method, as this method would gives underestimated height with a large variability.

The threshold of $\frac{d\theta}{dz}$ would be interesting to try too. I think I have seen it used in some works, just not in the ones I mention in this code.